Fix super long error output when test fails in Travis build

I'm not sure why, but lately my builds have been throwing the same error over and over again, sometimes running for more than 45 minutes.

For example, [this build](https://travis-ci.org/HabitRPG/habitrpg/builds/147853560) only stopped because the output file exceeded the limit of 4MB, with over 80k lines of output.  And this only happened after more than 48 minutes.

It's just this failure over and over again:

> 'build:dev' errored after 28 s
> Error in plugin 'run-sequence(test:karma)'
> Message:
>     Command failed: /bin/sh -c NODE_ENV=test  npm run test:karma

I think we should limit the number of retries for running karma tests to prevent this from happening again and hogging TravisCI resources.

It seems like whenever a test fails, that test's test suite gets rerun recursively forever.

[this build](https://travis-ci.org/HabitRPG/habitrpg/builds/156625146): **(49 min 5 sec)**
-  after the test suite with the failed test:
  -  `npm ERR! Failed at the habitica@3.35.0 test:common script 'mocha test/common --recursive'`
-  and then these messages over and over:
  -  `'test' errored after 48 s`
  -  `Error in plugin 'run-sequence(test:common)'`

[this build](https://travis-ci.org/HabitRPG/habitrpg/builds/155522869): **(49 min 6 sec)**
-  after failed test:
  -  `Error: Command failed: /bin/sh -c NODE_ENV=test  mocha test/api/v3/unit --recursive`
-  and then these messages over and over:
  -  `'build:dev' errored after 2.35 min`
  -  `Error in plugin 'run-sequence(test:api-v3:unit)'`

[this build](https://travis-ci.org/HabitRPG/habitrpg/builds/157452243): **()**
-  after failed test:
  -  `Error: Command failed: /bin/sh -c NODE_ENV=test  mocha test/api/v3/integration --recursive`
-  and then these messages over and over:
  -  `'test' errored after 7 min`
  -  `Error in plugin 'run-sequence(test:api-v3:integration)'`

However, if the build fails because of linting errors, it just stops and says the build failed:

[this build](https://travis-ci.org/HabitRPG/habitrpg/builds/157471241):  **(5 min 6 sec)**
-  `npm ERR! Failed at the habitica@3.38.0 lint script 'eslint .'.`

[this build](https://travis-ci.org/HabitRPG/habitrpg/builds/157397424):  **(5 min 11 sec)**
-  `npm ERR! Failed at the habitica@3.38.0 lint script 'eslint .'.`

[this build](https://travis-ci.org/HabitRPG/habitrpg/builds/157306542):  **(6 min 11 sec)**
-  `npm ERR! Failed at the habitica@3.38.0 lint script 'eslint .'.`

--- a/tasks/gulp-tests.js
+++ b/tasks/gulp-tests.js
@@ -90,7 +90,10 @@ gulp.task('test:sanity', (cb) => {
   let runner = exec(
     testBin(SANITY_TEST_COMMAND),
     (err, stdout, stderr) => {
-      cb(err);
+      if (err) {
+        process.exit(1);
+      }
+      cb();
     }
   );
   pipe(runner);
@@ -100,7 +103,10 @@ gulp.task('test:common', ['test:prepare:build'], (cb) => {
   let runner = exec(
     testBin(COMMON_TEST_COMMAND),
     (err, stdout, stderr) => {
-      cb(err);
+      if (err) {
+        process.exit(1);
+      }
+      cb();
     }
   );
   pipe(runner);
@@ -135,7 +141,10 @@ gulp.task('test:content', ['test:prepare:build'], (cb) => {
     testBin(CONTENT_TEST_COMMAND),
     CONTENT_OPTIONS,
     (err, stdout, stderr) => {
-      cb(err);
+      if (err) {
+        process.exit(1);
+      }
+      cb();
     }
   );
   pipe(runner);
@@ -196,7 +205,10 @@ gulp.task('test:karma', ['test:prepare:build'], (cb) => {
   let runner = exec(
     testBin(KARMA_TEST_COMMAND),
     (err, stdout) => {
-      cb(err);
+      if (err) {
+        process.exit(1);
+      }
+      cb();
     }
   );
   pipe(runner);
@@ -243,7 +255,10 @@ gulp.task('test:e2e', ['test:prepare', 'test:prepare:server'], (cb) => {
       'npm run test:e2e',
       (err, stdout, stderr) => {
         support.forEach(kill);
-        cb(err);
+        if (err) {
+          process.exit(1);
+        }
+        cb();
       }
     );
     pipe(runner);
@@ -282,7 +297,12 @@ gulp.task('test:e2e:safe', ['test:prepare', 'test:prepare:server'], (cb) => {
 gulp.task('test:api-v3:unit', (done) => {
   let runner = exec(
     testBin('mocha test/api/v3/unit --recursive'),
-    (err, stdout, stderr) => done(err)
+    (err, stdout, stderr) => {
+      if (err) {
+        process.exit(1);
+      }
+      done();
+    }
   )
 
   pipe(runner);
@@ -296,7 +316,12 @@ gulp.task('test:api-v3:integration', (done) => {
   let runner = exec(
     testBin('mocha test/api/v3/integration --recursive'),
     {maxBuffer: 500*1024},
-    (err, stdout, stderr) => done(err)
+    (err, stdout, stderr) => {
+      if (err) {
+        process.exit(1);
+      }
+      done();
+    }
   )
 
   pipe(runner);

